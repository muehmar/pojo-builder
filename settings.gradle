rootProject.name = "pojo-builder"

include 'core'
include 'core-annotations'
include 'example'
include 'example-java17'

dependencyResolutionManagement {
    versionCatalogs {
        libs {
            version('pojobuilder', '1.6.0')
            library('pojobuilder-core', 'io.github.muehmar','pojo-builder').versionRef('pojobuilder')
            library('pojobuilder-annotations', 'io.github.muehmar','pojo-builder-annotations').versionRef('pojobuilder')

            library('codeGenerator', 'io.github.muehmar:code-generator:0.22.0')

            version('junit', '5.10.0')
            library('junit-api', 'org.junit.jupiter', 'junit-jupiter-api').versionRef('junit')
            library('junit-engine', 'org.junit.jupiter', 'junit-jupiter-engine').versionRef('junit')
            library('junit-params', 'org.junit.jupiter', 'junit-jupiter-params').versionRef('junit')

            version('assertj', '3.26.3')
            library('assertj', 'org.assertj', 'assertj-core').versionRef('assertj')

            bundle('junit', ['junit-api', 'junit-engine', 'junit-params', 'assertj'])

            library('jakarta-annotationApi', 'jakarta.annotation:jakarta.annotation-api:2.1.1')

            library('google-autoservice', 'com.google.auto.service:auto-service:1.1.1')
            library('joor', 'org.jooq:joor-java-8:0.9.15')

            library('lombok', 'org.projectlombok:lombok:1.18.32')

            library('snapshotTesting','io.github.origin-energy:java-snapshot-testing-junit5:4.0.8')

            library('slf4j', 'org.slf4j:slf4j-api:2.0.7')

            plugin('release', 'net.researchgate.release').version('2.8.1')
            plugin('versions', 'com.github.ben-manes.versions').version('0.51.0')
        }
    }
}