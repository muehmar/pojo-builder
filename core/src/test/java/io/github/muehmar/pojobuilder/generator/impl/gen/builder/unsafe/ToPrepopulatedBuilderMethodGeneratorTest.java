package io.github.muehmar.pojobuilder.generator.impl.gen.builder.unsafe;

import static io.github.muehmar.codegenerator.writer.Writer.javaWriter;
import static io.github.muehmar.pojobuilder.generator.impl.gen.builder.unsafe.ToPrepopulatedBuilderMethodGenerator.toPrepopulatedBuilderMethodGenerator;
import static io.github.muehmar.pojobuilder.snapshottesting.SnapshotUtil.writerSnapshot;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.annotations.SnapshotName;
import io.github.muehmar.codegenerator.Generator;
import io.github.muehmar.codegenerator.writer.Writer;
import io.github.muehmar.pojobuilder.generator.Pojos;
import io.github.muehmar.pojobuilder.generator.model.Pojo;
import io.github.muehmar.pojobuilder.generator.model.settings.PojoSettings;
import io.github.muehmar.pojobuilder.snapshottesting.SnapshotTest;
import org.junit.jupiter.api.Test;

@SnapshotTest
class ToPrepopulatedBuilderMethodGeneratorTest {
  private Expect expect;

  @Test
  @SnapshotName("samplePojo")
  void toPrepopulatedBuilderMethodGenerator_when_samplePojo_then_correctOutput() {
    final Generator<Pojo, PojoSettings> generator = toPrepopulatedBuilderMethodGenerator();
    final Writer writer =
        generator.generate(Pojos.sample(), PojoSettings.defaultSettings(), javaWriter());

    expect.toMatchSnapshot(writerSnapshot(writer));
  }

  @Test
  @SnapshotName("genericSamplePojo")
  void generate_when_genericSamplePojo_then_matchSnapshot() {
    final Generator<Pojo, PojoSettings> generator = toPrepopulatedBuilderMethodGenerator();
    final Writer writer =
        generator.generate(Pojos.genericSample(), PojoSettings.defaultSettings(), javaWriter());

    expect.toMatchSnapshot(writerSnapshot(writer));
  }
}
