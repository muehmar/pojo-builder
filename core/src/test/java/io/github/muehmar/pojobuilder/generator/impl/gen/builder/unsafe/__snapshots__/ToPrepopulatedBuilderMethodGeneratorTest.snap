genericSamplePojo=[
.
.
/**
 * Creates a new instance of {@link CustomerBuilder} where it is guaranteed that
 * all required fields are already set.
 */
public CustomerBuilder<T, S> toBuilder() {
  return new CustomerBuilder<>(
      id,
      data,
      additionalData
  );
}
]


samplePojo=[
.
.
/**
 * Creates a new instance of {@link CustomerBuilder} where it is guaranteed that
 * all required fields are already set.
 */
public CustomerBuilder toBuilder() {
  return new CustomerBuilder(
      id,
      username,
      nickname
  );
}
]